helper.php

```diff
@@ -110,8 +110,22 @@ public function getSecret($service) {
      * @param $service
      * @return string
      */
+    public function getUrl($service) {
+        $service = strtolower($service);
+        return $this->getConf($service.'-url');
+    }
+
+    /**
+     * Return the configured Authentication Endpoint URL for the given service
+     *
+     * @param $service
+     * @return string
+     */
     public function getAuthEndpoint($service) {
         $service = strtolower($service);
+        if (strcmp($service, 'Gitlab')) {
+            return ($this->getUrl($service) . '/oauth/authorize');
+        }
```

Comment:
I'm not happy with these special cases. I feel like they should be part of the Adapter class.